front/src/reducers/osrdconf2/common/testUtils.ts

```diff
@@ -0,0 +1,674 @@
+        originTime: undefined,
+        originUpperBoundTime: '14:00:00',
+      });
+
+      store.dispatch(slice.actions.updateOriginUpperBoundTime('20:00:00'));
+
+      const state = store.getState()[slice.name];
+      expect(state.originTime).toBe('18:00:00');
+      expect(state.originUpperBoundTime).toBe('20:00:00');
+    });
+
+    it('should use default difference if originUpperBoundTime not defined', () => {
+      const store = createStore(slice, {
+        originLinkedBounds: true,
+        originUpperBoundTime: undefined,
+      });
+
+      store.dispatch(slice.actions.updateOriginUpperBoundTime('20:00:00'));
+
+      const state = store.getState()[slice.name];
+      expect(state.originTime).toBe('18:00:00');
+      expect(state.originUpperBoundTime).toBe('20:00:00');
+    });
+
+    it('upper bound should not go below lower bonud when unlinked', () => {
+      const store = createStore(slice, {
+        originLinkedBounds: false,
+        originTime: '14:00:00',
+        originUpperBoundTime: '18:00:00',
+      });
+
+      store.dispatch(slice.actions.updateOriginUpperBoundTime('12:00:00'));
+
+      const state = store.getState()[slice.name];
+      expect(state.originTime).toBe('14:00:00');
+      expect(state.originUpperBoundTime).toBe('14:00:00');
+    });
+  });
+
+  describe('should handle toggleOriginLinkedBounds', () => {
+    it('set to false if true ', () => {
+      const store = createStore(slice, { originLinkedBounds: true });
+      store.dispatch(slice.actions.toggleOriginLinkedBounds());
+      const state = store.getState()[slice.name];
+      expect(state.originLinkedBounds).toBe(false);
+    });
+
+    it('set to true if false ', () => {
+      const store = createStore(slice, { originLinkedBounds: false });
+
+      store.dispatch(slice.actions.toggleOriginLinkedBounds());
+      const state = store.getState()[slice.name];
+      expect(state.originLinkedBounds).toBe(true);
+    });
+  });
+
+  it('should handle updateOriginDate', () => {
+    const store = createStore(slice);
+    const newOriginDate = '13/12/2023';
+    store.dispatch(slice.actions.updateOriginDate(newOriginDate));
+    const state = store.getState()[slice.name];
+    expect(state.originDate).toBe(newOriginDate);
+  });
+
+  it('should handle updateOriginUpperBoundDate', () => {
+    const store = createStore(slice);
+    const newOriginUpperBoundDate = '13/12/2023';
+    store.dispatch(slice.actions.updateOriginUpperBoundDate(newOriginUpperBoundDate));
+    const state = store.getState()[slice.name];
+    expect(state.originUpperBoundDate).toBe(newOriginUpperBoundDate);
+  });
+
+  it('should handle replaceVias', () => {
+    const store = createStore(slice);
+    const newVias: PointOnMap[] = [testDataBuilder.buildPointOnMap()];
+    store.dispatch(slice.actions.replaceVias(newVias));
+    const state = store.getState()[slice.name];
+    expect(state.vias).toBe(newVias);
+  });
+
+  it('should handle updateVias', () => {
```

Comment:
If you rename updateVias -> addVia, don't forget to change the test description :)
```suggestion
  it('should handle addVias', () => {
```